test-data/unit/check-statements.test

```diff
@@ -2195,6 +2195,23 @@ class B: pass
 def foo(x: int) -> Union[Generator[A, None, None], Generator[B, None, None]]:
     yield x  # E: Incompatible types in "yield" (actual type "int", expected type "Union[A, B]")
 
+[case testReturnTypeOfGeneratorUnion]
+from typing import Generator, Union
+
+class T: pass
+
+def foo(arg: Union[Generator[int, None, T], Generator[str, None, T]]) -> Generator[Union[int, str], None, T]:
+    return (yield from arg)
+
+[case testYieldFromError]
+from typing import Generator, Union
+
+class T: pass
+class R: pass
+
+def foo(arg: Union[T, R]) -> Generator[Union[int, str], None, T]:
```

Comment:
Thanks! What about a Union involving a Generator and some other sort of iterable?

Reply:
I added a test for the union of a generator and iterable